File: src/br/com/globalcode/android/MyTabListener.scala
package br.com.globalcode.android

import android.app.ActionBar.Tab
import android.app.Activity
import android.app.Fragment
import android.app.FragmentTransaction

class MyTabListener[_](activity: Activity, tag: String, clz: Class[_]) extends
		android.app.ActionBar.TabListener  {

	var mFragment: Fragment = new Fragment();
	
	override def onTabSelected(tab: Tab, ft: FragmentTransaction) : Unit = {

		mFragment = activity.getFragmentManager().findFragmentByTag(tag);

		if (mFragment == null) {
			
			mFragment = Fragment.instantiate(activity, clz.getName());
			ft.add(android.R.id.content, mFragment, tag);
		} else {
			
			ft.attach(mFragment);
		}
	}

	override def onTabReselected(tab: Tab, ft: FragmentTransaction) = println("onTabReselected event")
	
	override def onTabUnselected(tab: Tab, ft: FragmentTransaction) : Unit = {
		
		if (mFragment != null) {
			ft.detach(mFragment);
		}
	}
}

File: src/br/com/globalcode/android/MyTabListener.java
package br.com.globalcode.android;

import android.app.ActionBar.Tab;
import android.app.Activity;
import android.app.Fragment;
import android.app.FragmentTransaction;

public class MyTabListener<T extends Fragment> implements
		android.app.ActionBar.TabListener {

	private Fragment mFragment;
	private final Activity mActivity;
	private final String mTag;
	private final Class<T> mClass;

	public MyTabListener(Activity activity, String tag, Class<T> clz) {
		mActivity = activity;
		mTag = tag;
		mClass = clz;
	}

	@Override
	public void onTabSelected(Tab tab, FragmentTransaction ft) {

		mFragment = mActivity.getFragmentManager().findFragmentByTag(mTag);

		if (mFragment == null) {
			
			mFragment = Fragment.instantiate(mActivity, mClass.getName());
			ft.add(android.R.id.content, mFragment, mTag);
		} else {
			ft.attach(mFragment);
		}
	}

	@Override
	public void onTabReselected(Tab tab, FragmentTransaction ft) {

	}

	@Override
	public void onTabUnselected(Tab tab, FragmentTransaction ft) {
		if (mFragment != null) {
			ft.detach(mFragment);
		}
	}

}
